test/unit/lib/plugins/aws/lib/monitor-stack.test.js

```diff
@@ -443,20 +444,21 @@ describe('monitorStack', () => {
       describeStackEventsStub.onCall(2).resolves(updateRollbackEvent);
       describeStackEventsStub.onCall(3).resolves(updateRollbackComplete);
 
-      return awsPlugin.monitorStack('update', cfDataMock, { frequency: 10 }).catch((e) => {
-        let errorMessage = 'An error occurred: ';
-        errorMessage += 'mochaS3 - Bucket already exists.';
-        if (e.name !== 'ServerlessError') throw e;
-        expect(e.name).to.be.equal('ServerlessError');
-        expect(e.message).to.be.equal(errorMessage);
-        expect(describeStackEventsStub.callCount).to.be.equal(4);
-        expect(
-          describeStackEventsStub.calledWithExactly('CloudFormation', 'describeStackEvents', {
-            StackName: cfDataMock.StackId,
-          })
-        ).to.be.equal(true);
-        awsPlugin.provider.request.restore();
-      });
+      await expect(awsPlugin.monitorStack('update', cfDataMock, { frequency: 10 }))
+        .to.eventually.be.rejectedWith(
+          ServerlessError,
+          'An error occurred: mochaS3 - Bucket already exists.'
+        )
+        .then(() => {
```

Comment:
Once we refactor it this way, ,let's not use `then`. Below logic can simply follow `awat expect..`

(and let's apply same in other refactored cases)

Reply:
alright, I agree

using `await` can reduce nesting and also avoid magical deeply nested promise chain